### ISSUE TITLE
iOS 10 notification not received

### ISSUE THREAD

**Issue by Hurinj** [phonegap/phonegap-plugin-push#1241]
Hello,

I just updated the os of my iphone to the iOS10.
Now I can't receive the notifications, but I tested with postman and the notif is successfuly sent : 
here is the response status 200 : 

```
{
  "multicast_id": 7144046489716950275,
  "success": 1,
  "failure": 0,
  "canonical_ids": 0,
  "results": [
    {
      "message_id": "0:1474819584882881%8ae8df6e8ae8df6e"
    }
  ]
}
```

Nothing is called on the javascript function `push.on('notification', function (data) { ... }`
I think it is not even fired.
I am using the FCM branch and everything was ok before. The installation of the app under iOS10 made it bug ;)

Do you know what can I do ?

**Comment by abdohossam5** [comment]
had the same issue and fixed by updating to the latest plugin version 1.8.2 and enabling push notification in xcode capabilities.

found this solution from issue #1154

**Comment by Bart-Holland** [comment]
If you are not using xcode, how can you add push notification in xcode capabilities?

**Comment by abdohossam5** [comment]
as per what is mentioned in xcode adding push notifications  capability does 2 things

1. add the push notification feature to your app id
2. add push notification entitlement to your entitlements file

I think you can do both steps manually if you don't want to use xcode.

**Comment by Sentence** [comment]
I found this problem too on iOS 10 with plugin version 1.8.2.

I can send push direclty with [Pusher application](https://github.com/noodlewerk/NWPusher), received correclty, but if i call it from standart APN, the device vibrate right, but not shows the message. Inside the app the receive function works.

Pusher sends:
`{"aps":{"alert":"Testing... (17)","badge":1,"sound":"default"}}`

And my PHP code sends:
`{"aps":{"alert":"Welcome!","badge":1,"sound":"default"}}`

I'll write if foudn the solution.

**Comment by Sentence** [comment]
Aaaand i found the iOS10's issue.

The iOS10 modified a lot ov things we need to send something different.

Pusher works with that thing, but I think the log doesn't updated yet.

I found an example payload [here](https://www.sitepoint.com/developing-push-notifications-for-ios-10/#notificationpayloads). Updated it and I send
`{"aps":{"alert":{"title":"Welcome", "body":"In this world"},"badge":1,"sound":"default"}}`

Received the title + message notification, opens the app and got the received event.
